### ISSUE TITLE
ExperimentSetTest.testExperimentRunsFindValuesBetween Fails Sometimes

### ISSUE THREAD

**Issue by thomasWeise** [optimizationBenchmarking/optimizationBenchmarking#1]
Under some circumstances, the test
`ExperimentSetTest.testExperimentRunsFindValuesBetween` in `RandomParallelTest` failes. In other words, finding a log point between two other log points seems to sometimes cause problems. It is always the (current) line 457: `Assert.assertSame(first, run.find(index, o));` which causes the error.

It could be that this is an issue with the test cases: Let's say we have two randomly generated values for a dimension, A and B. If they are of double precision and B>A, it may be that 0.5(A+B) equals either A or B, if both values are too close together. Thus, I try to modify the test case to prevent this.

Since I cannot reproduce the error and the unit test normally runs well, I will need to investigate the code again. For now, I will use the modified test and see if that error still occurs.

Failing Builds:
- http://email.circleci.com/c/eJx9j8FqxSAQRb_G7CLRSYwuXLSUtyj0G4rRSTK8aEKibenX1z66LgwX5hyY4QarldENWT0Ko0cADiMXPRijxWiMEj1_dR_uzdHGy1RSLqzv6GhF1w6tANlqwdFLTinjmdzWrFZMXnYOBmdA42x6PSjpB5B9kOMMHTSbXXM-LgZPTN7qeDr9hp6432Ndl7XGfmSK9O0y7ekZk1-jO--Ulv8VKMXgVnJ89y4ejpbE4GUqtIV2rg0wMKl-bcRAJVaHseI_eO3l9Fhh2jPN5B8PmtPmT6QLa-0lfvGATbaPi9cPpFxmTg
- https://semaphoreci.com/thomasweise/optimizationbenchmarking/branches/master/builds/255
- https://semaphoreci.com/thomasweise/optimizationbenchmarking/branches/master/builds/247

**Comment by thomasWeise** [comment]
With the better error information and a night of JUnit testing, I got the following traces:

```
Failed tests: 
  RandomParallelTest>ExperimentSetTest.testExperimentRunsFindValuesBetween:495
  The log points [-9.926444257218427E34, -5.7248705037915285E298, -1833442614, -2.78555443E9, 54]
  and
  [-9.931434420361298E34, -5.72484803362016E298, -1833228798, -2.78555469E9, 54]
  should be the same, but they are not.
  We looked for value -2.78555456E9 of dimension 3 (kkwqcyihiepvcd(QUALITY_PROBLEM_DEPENDENT,DECREASING,float))
  between the values -2.785554432E9 and -2.785554688E9 of points
  [-9.926444257218427E34, -5.7248705037915285E298, -1833442614, -2.78555443E9, 54]
  and
  [-9.931434420361298E34, -5.72484803362016E298, -1833228798, -2.78555469E9, 54]
  at indexes 5 and 6 of 16


Failed tests: 
  RandomParallelTest>ExperimentSetTest.testExperimentRunsFindValuesBetween:495
  The log points [-84, -112905595, -9.302471E14]
  and
  [-84, -112912675, -9.3024717E14]
  should be the same, but they are not.
  We looked for value -9.302471344128E14 of dimension 2 (ypzpmrhajvkzld(ITERATION_SUB_FE,INCREASING,float))
  between the values -9.30247167967232E14 and -9.30247100858368E14 of points
  [-84, -112912675, -9.3024717E14] and [-84, -112905595, -9.302471E14]
  at indexes 7 and 8 of 18
```

This brings me on the right path to solve this long-standing, sporadically occurring error: The values to search are generated as `double`s in my test, but the dimension is `float`.:
1. `-2.78555456E9` and `-2.785554432E9` are both equal to `-2.78555443E9`
2. `-9.302471344128E14` and `-9.30247100858368E14` are both equal to `-9.302471E14`

In other words, the code is working correctly, the error is likely in the JUnit test. I will now modify the test.
